src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -280,4 +283,17 @@ object Validation {
       validator(t)
     }
   }
+
+  /**
+    * A validator that succeeds only if the provided string matches the specified pattern.
+    *
+    * Initially took from com.wix.accord.combinators.StringCombinators to enhance validationMessage feature
+    */
+  class MatchesRegexWithMessage(pattern: Pattern, partialMatchAllowed: Boolean = true, validationMessage: String)
+    extends NullSafeValidator[String](
+      v => if (partialMatchAllowed) pattern.matcher(v).find() else pattern.matcher(v).matches(),
+      v => v -> validationMessage)
+
+  def matchRegexFullyWithMessage(regex: Regex, message: String): Validator[String] =
```

Comment:
I think you can boil this down to
```
  def regexValidator(regex: Regex, failureMessage: String) = new NullSafeValidator[String](
    test = _.matches(regex.regex),
    failure = _ -> failureMessage
  )
```
withouth the need for `MatchesRegexWithMessage `.